pkgs/checks/test/extensions/core_test.dart

```diff
@@ -0,0 +1,167 @@
+import 'package:checks/checks.dart';
+import 'package:checks/context.dart';
+import 'package:test/test.dart';
+
+Matcher isARejection({Object? which, Object? actual}) {
+  var rejection = isA<Rejection>().having((p0) => p0.which, 'which', which);
+
+  rejection = rejection.having((p0) => p0.actual, 'actual', actual);
+
+  return rejection;
+}
+
+void main() {
+  group('TypeChecks', () {
+    test('isA', () {
+      checkThat(1).isA<int>();
+
+      expect(
```

Comment:
We should not use `expect` or `Matcher` in these tests.

Reply:
To bootstrap it seemed a good idea – so we're not leaning on the thing we're testing.

Reply:
I disagree, that sounds like building a bunch of debt.

We had no problem using `expect` to test `expect`. I don't see what we'll have any difficulty here.

Reply:
fixed!